libr/core/cmd_anal.c

```diff
@@ -1220,7 +1222,12 @@ R_API char *cmd_syscall_dostr(RCore *core, int n) {
 		res = r_str_appendf (res, "%d = unknown ()", n);
 		return res;
 	}
-	res = r_str_appendf (res, "%d = %s (", item->num, item->name);
+
+	if (!strcmp (os, "macos")) {
```

Comment:
I'm not familiar with this part of the code, but could you tell us why it's necessary to special handling `macos` in all these parts of the code?

Reply:
mac uses large number hex syscalls, take a look at the sdb 
https://github.com/radare/radare2/blob/master/libr/syscall/d/darwin-x86-64.sdb.txt#L20

its not intuitive to talk about 536870939 or 536870934 syscall, but rather 0x2000001B or 0x20000016

Reply:
can't we just use always hex then?

Reply:
Thats up to you all

Reply:
Though note the hardcoding of linux's int 80 would still need to be switched on os for other platforms.